### PR TITLE
Add Volume Output to Video Layer Node

### PR DESCRIPTION
**Pull request by narner**
https://github.com/user-attachments/assets/53b99641-9c32-4f8f-9878-1983cd06d658

### CODE REVIEW

File: Stitch/Graph/Node/Layer/Type/VisualMediaLayerNode.swift [hunk 1]
```diff
@@ -188,6 +188,8 @@ struct VideoLayerView: View {
             shadowOffset: viewModel.shadowOffset.getPosition ?? .defaultShadowOffset,
             parentSize: parentSize,
             parentDisablesPosition: parentDisablesPosition,
-            isClipped: viewModel.isClipped.getBool ?? false)
+            isClipped: viewModel.isClipped.getBool ?? false,
+            volume: viewModel.volume.getNumber ?? 0.5
```
Comment (ellbosch) [inline Stitch/Graph/Node/Layer/Type/VisualMediaLayerNode.swift:192]: Does this match Origami's default volume?

File: Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift [hunk 2]
```diff
@@ -1,14 +1,7 @@
...
-//  Stitch
-//
-//  Created by Elliot Boschwitz on 8/5/22.
-//
-
 import AVKit
 
-/// A class for managing `AVPlayer` instances for non-layer scenarios. For video layers,
-/// `StitchVideoViewController` is used.
+var DEFAULT_VIDEO_PLAYER_VOLUME = 0.5
```
Comment (ellbosch) [inline Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift:3]: Please move this inside StitchVideoImportPlayer and make it both static and _constant_ please!

File: Stitch/Graph/Node/Layer/Type/VisualMediaLayerNode.swift (hunk 1 above)
Comment (narner) [reply Stitch/Graph/Node/Layer/Type/VisualMediaLayerNode.swift:192]: Ah no; there's is 1; good catch

File: Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift [hunk 3]
```diff
@@ -48,15 +39,14 @@ final class StitchVideoImportPlayer: Sendable {
             self?.thumbnail = await player?.currentItem?.asset.getThumbnail()
         }
 
-        // Video muted by default, only unmuted by video layer
-        self.muteSound()
+        self.muteSound() // Mute initially or customize if needed.
```
Comment (ellbosch) [inline Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift:42]: This looks redundant.

File: Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift [hunk 4]
```diff
@@ -71,6 +61,16 @@ final class StitchVideoImportPlayer: Sendable {
...
         self.stitchVideoDelegate.audio.delegate.volume
     }
+    
+    @MainActor
+    func setVolume(volume: Double) {
+        guard volume >= 0.0 && volume <= 1.0 else {
+            print("Volume must be between 0.0 and 1.0")
+            return
+        }
+        self.video.volume = Float(volume)
```
Comment (ellbosch) [inline Stitch/Graph/Node/Port/Model/PortValue/Type/Media/ObjectTypes/Video/StitchVideoPlayer.swift:71]:
```suggestion
        let volume = min(1, max(0, volume))
        self.video.volume = Float(volume)
```

File: Stitch/Graph/Node/Port/View/Field/InputView/ScrubbedVideoView.swift [hunk 5]
```diff
@@ -14,21 +14,28 @@ import StitchSchemaKit
...
     func makeUIViewController(context: Context) -> StitchVideoViewController {
         let vc = StitchVideoViewController(videoPlayer: videoPlayer)
         vc.updateVideoGravity(with: fitStyle)
 
         // Sound disabled until layer is created for video
         self.videoPlayer.enableSound()
-
+        self.videoPlayer.setVolume(volume: volume)
+        // Set volume when the view is created
+        videoPlayer.setVolume(volume: volume)
```
Comment (ellbosch) [inline Stitch/Graph/Node/Port/View/Field/InputView/ScrubbedVideoView.swift:27]:
```suggestion
        // Set volume when the view is created
        videoPlayer.setVolume(volume: volume)
```
```suggestion
        self.videoPlayer.setVolume(volume: volume)
        // Set volume when the view is created
        videoPlayer.setVolume(volume: volume)
```

**Review by ellbosch — CHANGES_REQUESTED** [review]
Good start. Please test this with video scrubbing to make sure we get expected behavior there.

**Review by ellbosch — APPROVED** [review]

**Pull request merged**